app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedActivity.kt

```diff
@@ -21,18 +21,29 @@ class RecentlyPlayedActivity : InjectableAppCompatActivity(), RouteToExploration
       RECENTLY_PLAYED_ACTIVITY_INTERNAL_PROFILE_ID_KEY,
       -1
     )
-    recentlyPlayedActivityPresenter.handleOnCreate(internalProfileId)
+    val recentlyPlayedTitleEnum =
+      intent.getSerializableExtra(RECENTLY_PLAYED_ACTIVITY_TITLE_ENUM_KEY)
+        as RecentlyPlayedTitleEnum
+    recentlyPlayedActivityPresenter.handleOnCreate(internalProfileId, recentlyPlayedTitleEnum)
   }
 
   companion object {
     // TODO(#1655): Re-restrict access to fields in tests post-Gradle.
     const val RECENTLY_PLAYED_ACTIVITY_INTERNAL_PROFILE_ID_KEY =
       "RecentlyPlayedActivity.internal_profile_id"
 
+    internal const val RECENTLY_PLAYED_ACTIVITY_TITLE_ENUM_KEY =
+      "RecentlyPlayedActivity.title_enum_key"
+
     /** Returns a new [Intent] to route to [RecentlyPlayedActivity]. */
-    fun createRecentlyPlayedActivityIntent(context: Context, internalProfileId: Int): Intent {
+    fun createRecentlyPlayedActivityIntent(
+      context: Context,
+      internalProfileId: Int,
+      recentlyPlayedTitleEnum: RecentlyPlayedTitleEnum
+    ): Intent {
       val intent = Intent(context, RecentlyPlayedActivity::class.java)
       intent.putExtra(RECENTLY_PLAYED_ACTIVITY_INTERNAL_PROFILE_ID_KEY, internalProfileId)
+      intent.putExtra(RECENTLY_PLAYED_ACTIVITY_TITLE_ENUM_KEY, recentlyPlayedTitleEnum)
```

Comment:
This seems reasonable to me, but I think we want to do a push toward using protos more aggressively for sending arguments & intent extras. Maybe just go ahead and define an intent extra proto for this activity & use that, instead? It means a less-clean enum but you could still create an extension function for ``getTitleFromStringRes`` to keep dependent code clean.

Reply:
I have tried to use protos. I am not sure if its correct so do let me know WDYT.

1. `DestinationScreen` will helps to identify which activity to open and also with what intent extra values.
2. For example for this implementation, `RecentlyPlayedActivityIntentExtras` is a value inside `DestinationScreen` which contains information about the destination activity as well as its intent extras.
3. In `activity` package I have created `ActivityRouter` which will be injected to all activities / fragments from where we want to start the next activity. In this we will call `routeToScreen` with correct `DestinationScreen` and then this will decide which activity should be started.
4. One added code clarity point: With this approach we can change `RecentlyPlayedActivityIntentExtras` proto anytime it will be very easy to make changes to code. For example in current develop as I had to send an enum I had to introduced a new key and change the `createRecentlyPlayedActivityIntent` function by introducing a new argument which in turn resulted in changes in all the instances where that function was getting called. But with this new approach these changes will be comparatively less.

Reply:
Ah, I was just thinking that you could pass a RecentlyPlayedActivityIntentExtras proto here & proceed similarly. The router approach seems much nicer. Do you want to do that change as part of this PR or separately?